### PR TITLE
[Defluent] Skip clone uses trait on DefluentReturnMethodCallRector

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: src/PhpParser/AstResolver.php [hunk 1]
```diff
@@ -340,15 +331,34 @@ public function resolvePropertyFromPropertyReflection(
         return $this->findPromotedPropertyByName($nodes, $desiredPropertyName);
     }
 
+    private function locateClassMethodInTrait(string $className, string $methodName): ?Node
```
Comment (TomasVotruba) [inline src/PhpParser/AstResolver.php]:
```suggestion
    private function locateClassMethodInTrait(string $className, string $methodName): ?ClassMethod
```

File: src/PhpParser/AstResolver.php [hunk 2]
```diff
@@ -340,15 +331,34 @@ public function resolvePropertyFromPropertyReflection(
...
      */
-    private function parseFileNameToDecoratedNodes(string $fileName): ?array
+    private function parseFileNameToDecoratedNodes(string $fileName): array
     {
         $fileContent = $this->smartFileSystem->readFile($fileName);
         $nodes = $this->parser->parse($fileContent);
         if ($nodes === null) {
-            return null;
+            return [];
         }
```
Comment (TomasVotruba) [inline src/PhpParser/AstResolver.php]: Why this change?

I would prefer to keep it, as we had lot of bugs with false positives like these, when we parsed a broken file. Instead of an error we returned nodes with assumption they should be iterated and file is editable.

Returning `null` kills this possibility right in the start.

Comment (samsonasik) [reply src/PhpParser/AstResolver.php]: That was to remove unneeded check when call it, used in `findFirst()` and `findInstanceOf()` which pass an array, pass empty array will make return null, which should be ok. 

Ok, I reverted it https://github.com/rectorphp/rector-src/pull/558/commits/d9311e71371b2809c8a756ebd1667a7a0866b76a

File: src/PhpParser/AstResolver.php (hunk 1 above)
Comment (samsonasik) [reply src/PhpParser/AstResolver.php]: updated.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thank you 😊
